### PR TITLE
Fix stopping the TrajectoryExecutionManager's execution

### PR DESCRIPTION
**Pull request by JafarAbdi**
### Description

Replace: https://github.com/ros-planning/moveit2/pull/493
Fix: https://github.com/ros-planning/moveit2/issues/503

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

**Comment by JafarAbdi** [comment]
@vimior I can't add you as a reviewer, could you please test this .?

**Comment by AndyZe** [comment]
I don't think this works. I tested with this branch:

https://github.com/AndyZe/moveit2_tutorials/tree/andyz/move_group_in_depth

With these commands:

`ros2 launch moveit2_tutorials move_group.launch.py`

`ros2 launch moveit2_tutorials move_group_interface_tutorial.launch.py`

The bit of code I added here should stop the robot, right?

```
  RCLCPP_ERROR(LOGGER, "Prepare to press Enter to stop the robot!");
  prompt("Press 'ENTER' to execute");
  move_group.asyncExecute(my_plan);
  prompt("Press 'Enter' to stop motion");
  move_group.stop();
```

**Comment by AndyZe** [comment]
Wait, it does work. Nice. :+1:

**Comment by AndyZe** [comment]
Question: what is the difference between `move_action_client_` and `trajectory_event_publisher_`?

Why do we have both?

File: moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp [hunk 1]
```diff
@@ -166,8 +166,15 @@ void TrajectoryExecutionManager::initialize()
 
   // other configuration steps
   reloadControllerInformation();
+  // The default callback group for rclcpp::Node is MutuallyExclusive which mean we can not call
```
Comment (AndyZe) [inline moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp:169]:
```suggestion
  // The default callback group for rclcpp::Node is MutuallyExclusive which means we cannot call
```

File: moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp [hunk 2]
```diff
@@ -166,8 +166,15 @@ void TrajectoryExecutionManager::initialize()
...
+  // receiveEvent while processing a different callback, to fix this we create a new callback group (the type is not
```
Comment (AndyZe) [inline moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp:170]:
```suggestion
  // receiveEvent while processing a different callback. To fix this we create a new callback group (the type is not
```

**Comment by JafarAbdi** [comment]
> Question: what is the difference between `move_action_client_` and `trajectory_event_publisher_`?
> 
> Why do we have both?

Not 100% sure, but I think `trajectory_event_publisher_` was meant to be used as an e-stop

File: moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp [hunk 3]
```diff
@@ -166,8 +166,15 @@ void TrajectoryExecutionManager::initialize()
...
+  // important since we only use it to process one callback) and associate event_topic_subscriber_ with this callback group
+  auto callback_group = node_->create_callback_group(rclcpp::CallbackGroupType::MutuallyExclusive);
+  auto options = rclcpp::SubscriptionOptions();
+  options.callback_group = callback_group;
   event_topic_subscriber_ = node_->create_subscription<std_msgs::msg::String>(
-      EXECUTION_EVENT_TOPIC, 100, std::bind(&TrajectoryExecutionManager::receiveEvent, this, std::placeholders::_1));
+      EXECUTION_EVENT_TOPIC, 100, std::bind(&TrajectoryExecutionManager::receiveEvent, this, std::placeholders::_1),
+      options);
```
Comment (AndyZe) [inline moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp:177]: This seems like a bizarre way to fix things. Oh, ROS2...

**Review by AndyZe — APPROVED** [review]
I'll approve this because it fixes an important bit of functionality. But, I made an issue to follow up with simplifying it. #507

**Comment by vimior** [comment]
> @vimior I can't add you as a reviewer, could you please test/review this .?

it does work. Nice. +1

**Pull request merged**